tests/python/unittest/test_io.py

```diff
@@ -161,13 +161,23 @@ def test_NDArrayIter_csr():
     shape = (num_rows, num_cols)
     csr, _ = rand_sparse_ndarray(shape, 'csr')
     dns = csr.asnumpy()
-    #test CSRNDArray with shuffle=True will throw NotImplementedError 
+
+    # CSRNDArray with last_batch_handle not equal to 'discard' will throw NotImplementedError 
     try:
-        csr_iter = mx.io.NDArrayIter({'data': csr}, dns, batch_size, shuffle=True,
```

Comment:
Let's use assertRaises to make it cleaner https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/common.py#L27